sound/effect.h

```diff
@@ -231,9 +231,31 @@ class Effect {
     selected_++;
     if (selected_ == (int)files_found()) selected_ = 0;
   }
+#ifdef NO_REPEAT_RANDOM
+  int last_selected_;
+#endif
   void SelectFloat(float value) {
     int f = files_found();
     selected_ = clamp(floorf(f * value), 0, f - 1);
+#ifdef NO_REPEAT_RANDOM
+    if (selected_ == last_selected_) {
+      if (f == 2) {
+        selected_ = rand() % num_files_;
```

Comment:
just se selected_ to -1 and let RandomFile do the picking.

Reply:
done